eclass/freebsd.eclass

```diff
@@ -95,6 +95,23 @@ freebsd_get_bmake() {
 	echo "${bmake}"
 }
 
+# Generates a patch SRC_URI or DISTDIR of upstream.
+freebsd_upstream_patches() {
+	local opt=$1
+	[[ ${#UPSTREAM_PATCHES[@]} -eq 0 ]] && return 1
+	for x in ${UPSTREAM_PATCHES[@]}
+	do
+		local out=${x/\//-}
+		out=${out/:/}
+		out="${PN}-${RV}-${out}"
+		if [[ ${opt} == -s ]] ; then
+			echo "${DISTDIR}/${out}"
```

Comment:
I think you mean plain `${out}` here. For in-DISTDIR files, we specify just the name.